test/mocks/azureClient.js

```
@@ -0,0 +1,140 @@
+      container: 'samplecontainer'
+    }
+  }
+};
+const provider = config.backup.provider;
+const objectStoreUrl = `https://${provider.storageAccount}.blob.core.windows.net`;
+
+exports.auth = auth;
+exports.download = download;
+exports.upload = upload;
+exports.headObject = headObject;
+exports.getContainer = getContainer;
+exports.list = list;
+exports.remove = remove;
+exports.deleteSnapshot = deleteSnapshot;
+exports.config = config;
+
+function encodePath(pathname) {
+  return pathname.replace(/:/g, '%3A');
+}
+
+function auth() {
+  return nock('https://login.microsoftonline.com')
+    .replyContentLength()
+    .post(`/${provider.tenant_id}/oauth2/token?api-version=1.0`, body => _.isObject(body))
+    .reply(200, '{\"token_type\":\"Bearer\",\"expires_in\":\"3600\",\"expires_on\":\"1462557169\",\"not_before\":\"1462553269\",\"resource\":\"https://management.core.windows.net/\",\"access_token\":\"eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiIsIng1dCI6Ik1uQ19WWmNBVGZNNXBPWWlKSE1iYTlnb0VLWSIsImtpZCI6Ik1uQ19WWmNBVGZNNXBPWWlKSE1iYTlnb0VLWSJ9.eyJhdWQiOiJodHRwczovL21hbmFnZW1lbnQuY29yZS53aW5kb3dzLm5ldC8iLCJpc3MiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwiaWF0IjoxNDYyNTUzMjY5LCJuYmYiOjE0NjI1NTMyNjksImV4cCI6MTQ2MjU1NzE2OSwiYXBwaWQiOiJiOWU2ZTA3Yi1jNDNlLTQ3MzEtODVjYS05ODE3ODkyNzI0Y2QiLCJhcHBpZGFjciI6IjEiLCJpZHAiOiJodHRwczovL3N0cy53aW5kb3dzLm5ldC83MmY5ODhiZi04NmYxLTQxYWYtOTFhYi0yZDdjZDAxMWRiNDcvIiwib2lkIjoiNGUwNDNmODYtYjMzZC00YzNiLThjNTYtNWM3NTkyOGEzNzBlIiwic3ViIjoiNGUwNDNmODYtYjMzZC00YzNiLThjNTYtNWM3NTkyOGEzNzBlIiwidGlkIjoiNzJmOTg4YmYtODZmMS00MWFmLTkxYWItMmQ3Y2QwMTFkYjQ3IiwidmVyIjoiMS4wIn0.rROtJoq7sd0BIfLtS-Ra9-xIN9lQPXnN0NvR8BWgEP8imEp1M3ryN8v1IUldiQeb6yhsf0Jg1QqA_vI8HUvCpDoCA7MvmSrMCVfrY7hKFDl9cJeFMILVXjWRbCHp29k0A_pz1r8au07_MmGGrEpBpc0Z5P1rr9qwOe4SmLCuC1poyNuERpwMtHXS4MwJ7mjz9OHsn_pvVxV9TMA-lcBxqfsxWPf8kbmHiFeyLonBftD-h0X7wjwJ-EGK6WSQ-bBtWyJNyJ6sDxqSUB4WpuywVVe_pCPcnKLhqD5tbW_thkyQG7nJ4bM_qqXvKIEtgxO0HGDIA7KS5AdMgBx8uNAC7g\"}', {
```

are the mock response params expires_on, not_before used? if not, then remove them. If they are used then certainly hardcoding them might make tests fail later.

---

done